### DISCUSSION TITLE
HelmRelease .spec.chart.spec missing .values

### DISCUSSION THREAD

**Discussion by DevopsCareer** [fluxcd/flux2#3140] [Q&A]
Hello, I am using `helm.toolkit.fluxcd.io/v2beta1` but I have noticed that one can only use `.valuesFiles` but one can not use `.values` in `.spec.chart.spec` of HelmChartRelease, what is alternative if one just want to use a single file to override default values.yaml
eg : 

```values:
    replicaCount: 2

**Comment by kingdonb** [comment]
`valuesFiles` is a part of the Chart because the values files are embedded in the chart, and their packaging is part of the "chart" artifact itself – there can be more than one `values.yaml` file in a chart repository, and you could select a different one, such as `production-values.yaml` – an example [here](https://fluxcd.io/flux/components/source/helmcharts/#values-files) shows how they can be mixed and matched, or stacked for merging in a particular order.

`values` are a part of the release because they are fixed overrides for a particular installation that override the Chart's provided values. This is intentionally part of the Release and not part of the Chart. Hope this helps.

If you want to provide a single value and make it external to the release.yaml you can use a configmap or secret with `valuesFrom` which is part of the release. The release cannot take in a `values.yaml` file at reconcile time because it lives in etcd, it does not have the context of the filesystem or repository path where it was installed from like a helm repository has within itself at reconcile time.

https://fluxcd.io/flux/guides/helmreleases/#refer-to-values-in-configmap-and-secret-resources

You can find a reference about how to do this here^
